Sources/GRPCCodeGen/Pipeline.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+struct Pipeline<Input> {
+  typealias ParsedInput = CodeGenerationRequest
+  typealias Output = SourceFile
+
+  var parseInputStage: PipelineStage<Input, CodeGenerationRequest>
+  var generateCodeStage: PipelineStage<CodeGenerationRequest, Output>
```

Comment:
We should avoid this - I've left a comment on `PipelineStage.swift` about it.

Reply:
The purpose of the Pipeline is to give the user the option to generate the code for an IDL file by simply calling a function and plugging in the parser, and not having to worry about the other parts of the CodeGen. However, the PipelineStages seem a bit unnecessary, so I will probably get rid of them.